kernel/debugmalloc.c

```diff
@@ -393,10 +414,39 @@ int myst_debug_posix_memalign(void** memptr, size_t alignment, size_t size)
     const size_t block_size = _calculate_block_size(alignment, size);
     void* block = NULL;
     header_t* header = NULL;
+    size_t rsize = _round_up_to_multiple(size, sizeof(uint64_t));
```

Comment:
Why do we need to align the data size to 8 bytes?

Reply:
The data is padded at the end up to an 8-byte boundary. This forces the footer to be aligned on an 8-byte boundary.